### ISSUE TITLE
Strange behavior with initial cache files

### ISSUE THREAD

**Issue by rampartisan** [scikit-build/scikit-build#471]
Hey,

I am having quite a confusing time using -C to set an initial cache files. When running the command:

```
python3 setup.py build -- "-Cvariant/host-clang/release.cmake"

loading initial cache file variant/host-clang/release.cmake
CMake Error: Error processing file: /home/user/repos/project/_skbuild/linux-x86_64-3.7/cmake-build/variant/host-clang/release.cmake
```

CMake seems to be prefixing with the build dir? I can fix this by running:

`python3 setup.py build -- "-C../../../variant/host-clang/release.cmake"

But this is not an optimal setup, does anyone have an idea on what is going on?

sckit-build 0.10.0
python 4.7.3
`cmake 3.16.2

**Comment by thewtex** [comment]
The initial cache format is picky -- it should not just be a copied *CMakeCache.txt* file. Per [-C docs](https://cmake.org/cmake/help/v3.17/manual/cmake.1.html#options):

> The given file should be a CMake script containing set() commands that use the CACHE option, not a cache-format file.

**Comment by rampartisan** [comment]
We basically use this cache to set up some toolchains, it works just running cmake, so I do not think the contents of the file is the issue.Containing the following 

```
include("${CMAKE_CURRENT_LIST_DIR}/../../externals/rtcore/external/variant/host-clang/release.cmake")
include("${CMAKE_CURRENT_LIST_DIR}/../../externals/rtcore/external/cmake/known_toolchain.cmake")
```

It seems that cmake is trying to load some totally wrong path:

`CMake Error: Error processing file: /home/user/repos/project/_skbuild/linux-x86_64-3.7/cmake-build/variant/host-clang/release.cmake`

it should be:

` /home/user/repos/project/variant/host-clang/release.cmake
`

Why does this happen running with scikit-build and not cmake, is the question :p

**Comment by thewtex** [comment]
How about replacing `CMAKE_CURRENT_LIST_DIR` with something that should be more well defined, e.g. `CMAKE_BINARY_DIR` or `CMAKE_SOURCE_DIR`?

**Comment by rampartisan** [comment]
Good idea, no joy ;) the problem is not in the actual cache file, its where cmake is looking for that file. Using scikit-build cmake is for some weird reason receiving any relative paths passed to -C prefixed with 

`_skbuild/linux-x86_64-3.7/cmake-build/ `

if I pass an absolute path to -C, it also works fine. If I use the relative path with vanilla cmake, it works fine. If I use an relative path with scikit-build, the cmake-build directory gets prefixed. I have had a quick look through the source and can not work out why.

**Comment by thewtex** [comment]
Offhand, I do not recall the precedence of `-D` and `-C`, but another approach to try is to use a specified variable with an absolute path, e.g.

```
-DEXTERNALS_DIR=/an/absolute/path
```

with 

```
include("${EXTERNALS_DIR}/rtcore/external/variant/host-clang/release.cmake")
include("${EXTERNALS_DIR}/rtcore/external/cmake/known_toolchain.cmake")
```

**Comment by henryiii** [comment]
scikit-build (classic) after #1185 is now a thin wrapper around scikit-build-core's setuptools plugin. Closing all open issues; if this still applies, please reopen or open an issue on scikit-build-core. Thanks!
